server/src/repositories/shared-link.repository.ts

```diff
@@ -19,6 +19,95 @@ export type SharedLinkSearchOptions = {
 export class SharedLinkRepository {
   constructor(@InjectKysely() private db: Kysely<DB>) {}
 
+  @GenerateSql({ params: [DummyValue.UUID, DummyValue.STRING] })
+  getBySlug(userId: string, slug: string) {
```

Comment:
Instead of whatever this is right now, can't you just get the shared link entity that belongs to the slug (if present) and do everything with that?

Reply:
> `getBySlug` is used by the auth.service during `validateSharedLink`

That is... not true? 👀 
It's used in the shared link service.